code/modules/mob/living/simple_animal/simple_animal_vr.dm

```diff
@@ -44,21 +44,25 @@
 		var/obj/belly/B = belly
 		for(var/mob/living/M in B)
 			new_fullness += M.size_multiplier
-		new_fullness = round(new_fullness, 1) // Because intervals of 0.25 are going to make sprite artists cry.
+	new_fullness = round(new_fullness, 1) // Because intervals of 0.25 are going to make sprite artists cry.
 	vore_fullness = min(vore_capacity, new_fullness)
 
-/mob/living/simple_animal/update_icon()
-	. = ..() // Call sideways "parent" to decide state
+/mob/living/simple_animal/proc/update_vore_icon()
 	if(vore_active)
 		update_fullness()
 		if(!vore_fullness)
```

Comment:
I'd remove the if(vore_active) test here, make update_fullness() always just return early if(!vore_active), and then your if(!vore_fullness) here will always return and prevent vore icons on those without vore_active (becauase it will forever be 0). Then you can un-indent this entire proc.